### ISSUE TITLE
TextBox keyboard navigation: keys for next/previous word places caret at the wrong position

### ISSUE THREAD

**Issue by PermanentNovice** [AvaloniaUI/Avalonia#14726]
## Describe the bug

In a TextBox, pressing the control + left/right keys to jump to the next/previous word places the caret at the end/beginning of the current word, not the next/previous word.
## To Reproduce

1. In a TextBox, type "Now is the time." without the quotes.
2. Press the home button. The caret is to the left of the 'N' in the word 'Now'.
3. Press control + the right key. The caret is now to the right of the 'w' in the word 'Now'.

## Expected behavior

After step 3, the caret should be to the left of the 'i' in the word 'is'. The caret should have moved one more position to the right.

This is the behavior I am accustomed to. Is it different for locales other than the U.S.?
## Environment 

- OS: Windows 11
- Avalonia-Version: 11.09

**Comment by PermanentNovice** [comment]
To properly behave like other text input controls, the control + left/right keys for jumping words should not skip punctuation marks or symbols. For example, assume the box contains the phrase "Sir, are you sure?" (without quotes). If the caret is at index zero (to the left of the "S") and the user presses control + right, the caret should move between the "r" and the comma in the word "Sir," Pressing control + right again should move the caret to the left of the "a".

**Comment by Gillibald** [comment]
I would follow the `<TextArea/>` implementation here

**Comment by PermanentNovice** [comment]
Yes - a <textarea>. I am typing this on the Avalonia issues page on github.com in what I assume is a <textarea> input field. It behaves the way I described in the original post. I am blind and use a screen reader and navigate with a keyboard only. For me, this is an issue of consistent accessibility behavior.

**Comment by Gillibald** [comment]
Thanks for making us aware of this issue. I will adjust the keyboard navigation as you suggested.

**Comment by PermanentNovice** [comment]
Thanks. If anyone is curious, they might be interested to know that, when jumping from word to word with control + right/left, a screen reader speaks the entire word after the caret, not just the next letter. However, when the caret stops at a punctuation symbol, it speaks the symbol only.
